### ISSUE TITLE
Please add Districts

### ISSUE THREAD

**Issue by Knaven** [8manos/wc-city-select#26]
Can you edit and add districts after the dropdown of city?

**Comment by Mantish** [comment]
Hi,
This is not straightforward, but it's possible.
I personally cannot implement this now, so I'll just tag the issue for now

**Comment by vbernuy** [comment]
Hi, Mantish. Thank you very much for your plugin, it's amazing . I would like to know if it is possible to use the optgroup function for cities. So we could use the optgroup label as City and the optgroup options as Districts.

Attached reference image.

And many thanks again.
![optgroup for cities](https://user-images.githubusercontent.com/42545597/44348346-c0e6ce00-a45f-11e8-9f3a-32dc08c3f018.jpg)

**Comment by Mantish** [comment]
Hi, I'm not really familiar with that feature.
Is there documentation about it?

**Comment by rwkyyy** [comment]
@Mantish A city (larg one) can be divided in districts, for example:
**Bucharest** (Romania) has **6 districts** (named sectors).
**It will need** a _new input field_ (checkout), _option page_ with feature disable by default and a _list of districts_ separately? This **might need** a rework of the structures of the cities per country since a district is part of a city.

For example:
```
'B'=array(
        'Bucuresti' => array(
                'Sector 1',
                'Sector 2',
                'Sector 3',
                // onward
        ),
);
```
#33 further example comparision

**Comment by Mantish** [comment]
yeah, it needs quite a few changes in many places. Is that something you would be able to work on @rwkyyy ?

**Comment by rwkyyy** [comment]
@Mantish tell me what you need and I'll see what I can do.

**Comment by Mantish** [comment]
Well, basically we need to implement this whole feature:

- Define `$cities` object structure to support districts
- Add new field for cart, checkout and admin pages
- Populate districts field according to the selected city
- Options page to enable this feature? or maybe auto-enabled if present in the `$cities` object

**Comment by rwkyyy** [comment]
OK, will try to make room for some of these next week, my schedule is pretty loaded this week.

**Comment by rwkyyy** [comment]
@Mantish I've did some digging these past few weeks (luckly one of our clients wanted this already) and apparently -- for better integration with shipping operators, it's best to use the field CITY as a district selection (or at least for Romania) because Bucharest (capital) has 6 districts and nothing else -- when the shop owner generates an AWB with the integration of the shipping operator, it will take automatically the cities as a district without a problem.

I haven't been able to test this with multiple shipping operators, in Romania, just one, but I'll try to find the time this week to test with at least the 3 major ones and confirm it.

p.s. the shipping operator plugin does get & post of the default WC fields for shipping.
